### PR TITLE
fix: fix scene switching bug

### PR DESCRIPTION
**Pull request by JensSteenmetz**
## Description
This PR proposes a fix for [issue AP-156](https://zomsa.atlassian.net/browse/AP-156?atlOrigin=eyJpIjoiODk4Njk3OTUzYTcyNGJhMzgwMWQ4ODk0MDU2OTgwYzUiLCJwIjoiaiJ9). When going from game to menu to game, the game broke. You couldn't move the player, interact with the enemies, or move the camera. Now you can.

## Testability
Try switching back and forth between the main menu scene (Settings) and the game scene (InGameSettings) through the pause menu.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes (used the Settings and InGameSettings scenes)

### CODE REVIEW

**Review by SSelinn — DISMISSED** [review]
IT WORKS! :D

File: aplib.net-demo/Assets/Scripts/InputManager.cs [hunk 1]
```diff
@@ -29,18 +28,22 @@ public class InputManager : MonoBehaviour
...
     {
         if (Instance != null && Instance != this)
+        {
             Destroy(gameObject);
-        else
-            Instance = this;
-        DontDestroyOnLoad(gameObject);
+            return;
+        }
+        else Instance = this;
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Scripts/InputManager.cs:35]:
```suggestion
        else 
        {
            Instance = this;
        }
```

**Review by Tboefijn — CHANGES_REQUESTED** [review]
Left a nitpick comment

**Review by Tboefijn — APPROVED** [review]
LGTM!

**Review by SSelinn — APPROVED** [review]
yeye

**Pull request merged**